src/libm-tester/tester.c

```diff
@@ -124,3202 +113,4205 @@ uint64_t d2u(double d) {
-      for(j=0;j<sizeof(ya)/sizeof(double) && success;j++) {
-	if (!xisnan(child_pow(xa[i], ya[j]))) {
-	  success = false;
-	  break;
+      for(i=0;i<sizeof(xa)/sizeof(double) && success;i++) {
+	for(j=0;j<sizeof(ya)/sizeof(double) && success;j++) {
+	  if (child_pow(xa[i], ya[j]) != NEGATIVE_INFINITY) success = 0;
 	}
       }
-    }
 
-    showResult(success);
-  }
+      showResult(success);
+    }
 
-  {
-    fprintf(stderr, "If x is +0, and y is an odd integer greater than 0, the result is +0 ... ");
+    {
+      fprintf(stderr, "If x is negative infinity, and y greater than 0 and not an odd integer, the result is positive infinity : ");
 
-    double xa[] = { +0.0 };
-    double ya[] = { 1, 3, 5, 7, 100001 };
+      double xa[] = { NEGATIVE_INFINITY };
+      double ya[] = { 0.5, 1.5, 2, 2.5, 3.5, 4, 100000, 100000.5 };
 
-    boolean success = true;
-    for(i=0;i<sizeof(xa)/sizeof(double) && success;i++) {
-      for(j=0;j<sizeof(ya)/sizeof(double) && success;j++) {
-	if (!isPlusZero(child_pow(xa[i], ya[j]))) {
-	  success = false;
-	  break;
+      for(i=0;i<sizeof(xa)/sizeof(double) && success;i++) {
+	for(j=0;j<sizeof(ya)/sizeof(double) && success;j++) {
+	  if (child_pow(xa[i], ya[j]) != POSITIVE_INFINITY) success = 0;
 	}
       }
+
+      showResult(success);
     }
 
-    showResult(success);
-  }
+    {
+      fprintf(stderr, "If x is positive infinity, and y less than 0, the result is +0 : ");
 
-  {
-    fprintf(stderr, "If x is -0, and y is an odd integer greater than 0, the result is -0 ... ");
-
-    double xa[] = { -0.0 };
-    double ya[] = { 1, 3, 5, 7, 100001 };
-
-    boolean success = true;
-    for(i=0;i<sizeof(xa)/sizeof(double) && success;i++) {
-      for(j=0;j<sizeof(ya)/sizeof(double) && success;j++) {
-	double test = child_pow(xa[i], ya[j]);
-	if (!isMinusZero(test)) {
-	  fprintf(stderr, "xa = %.20g, ya = %.20g, test = %.20g, correct = %.20g\n", xa[i], ya[j], test, -0.0);
-	  success = false;
-	  break;
+      double xa[] = { POSITIVE_INFINITY };
+      double ya[] = { -100000.5, -100000, -3, -2.5, -2, -1.5, -1.0, -0.5 };
+
+      for(i=0;i<sizeof(xa)/sizeof(double) && success;i++) {
+	for(j=0;j<sizeof(ya)/sizeof(double) && success;j++) {
+	  if (!isPlusZero(child_pow(xa[i], ya[j]))) success = 0;
 	}
       }
-    }
 
-    showResult(success);
-  }
+      showResult(success);
+    }
 
-  {
-    fprintf(stderr, "If x is 0, and y greater than 0 and not an odd integer, the result is +0 ... ");
+    {
+      fprintf(stderr, "If x is positive infinity, and y greater than 0, the result is positive infinity : ");
 
-    double xa[] = { +0.0, -0.0 };
-    double ya[] = { 0.5, 1.5, 2.0, 2.5, 4.0, 100000, 100000.5 };
+      double xa[] = { POSITIVE_INFINITY };
+      double ya[] = { 0.5, 1, 1.5, 2.0, 2.5, 3.0, 100000, 100000.5 };
 
-    boolean success = true;
-    for(i=0;i<sizeof(xa)/sizeof(double) && success;i++) {
-      for(j=0;j<sizeof(ya)/sizeof(double) && success;j++) {
-	if (!isPlusZe
```

Comment:
How much of this could be folded into macros? I see tens of loops here that could be written with macros just calling the proper function names... 

Something like the following would save a lot of typing.

```
#ifdef TEST_01
#undef TEST_01
#define TEST_01(name, mpfr_function, child_function) // macro implementation for the first kind of loop
TEST_01("hypot_u35", mpfr_hypot, child_hipot_u35);
TEST_01("hypot_u05", mpfr_hypot, child_hipot_u05);
#undef TEST_01
#endif  // #ifdef TEST_01

#ifdef TEST_02
#undef TEST_02
#define TEST_02(name, mpfr_function, child_function) // macro implementation for another kind of loop
TEST_02("pow", mpfr_pow, child_pow);
// ... 
#undef TEST_02
#endif  // #ifdef TEST_02

```